FLT/Mathlib/RepresentationTheory/Basic.lean

```diff
@@ -0,0 +1,40 @@
+import Mathlib.LinearAlgebra.Matrix.GeneralLinearGroup.Defs
+import Mathlib.RepresentationTheory.Basic
+
+open LinearMap
+open scoped TensorProduct
+
+namespace Representation
+
+variable {R V G ι: Type*} [CommRing R] [AddCommMonoid V] [Module R V] [Module.Free R V]
+  [Module.Finite R V] [Group G] [DecidableEq ι] [Fintype ι]
+
+variable (ρ : Representation R G V) (𝓑 : Basis ι R V)
+
+omit [Module.Free R V] [Module.Finite R V] in
+@[simp]
+lemma comp_eq_mul (g h : G): ρ g ∘ₗ ρ h = ρ (g * h) := by
+  simp_all only [map_mul]
+  rfl
+
+noncomputable def gl_map_of_representation_of_basis
+  : G →* Matrix.GeneralLinearGroup ι R where
+    toFun g := {
+      val := LinearMap.toMatrix 𝓑 𝓑 (ρ g)
+      inv := LinearMap.toMatrix 𝓑 𝓑 (ρ g⁻¹)
+      val_inv := by rw [← toMatrix_comp, comp_eq_mul]; simp
+      inv_val := by rw [← toMatrix_comp, comp_eq_mul]; simp
+    }
+    map_one' := by aesop
+    map_mul' := by rintro x y; simp [LinearMap.toMatrix_mul]; norm_cast
+
+noncomputable def tprod' (R' : Type*) [CommRing R'] [Algebra R R'] (ρ : Representation R G V)
+    : Representation R G (R' ⊗[R] V) where
```

Comment:
Did you really mean to make R' (x)_R V an R-representation rather than R'-representation here? If you meant it to be an R'-representation then it could be called baseChange or something like that. If you did want it to be an R-representation then my instinct is to make baseChange anyway, and also restrictScalars from R'-reps to R-reps if R' is an R-algebra, and get tprod' by composing.